src/shared/gpt.c

```diff
@@ -35,10 +30,11 @@ static const GptPartitionType gpt_partition_type_table[] = {
         { GPT_VAR,                   "var"                   },
         { GPT_TMP,                   "tmp"                   },
         { GPT_LINUX_GENERIC,         "linux-generic",        },
+        {}
 };
 
 const char *gpt_partition_type_uuid_to_string(sd_id128_t id) {
-        for (size_t i = 0; i < ELEMENTSOF(gpt_partition_type_table); i++)
+        for (size_t i = 0; i < ELEMENTSOF(gpt_partition_type_table) - 1; i++)
```

Comment:
sounds weird using two methods for detecting the end of the list here... maybe just check for  `gpt_partition_type_table[i].name` being non-`NULL` instead here to always use the same condition check, both int code in this file and in external code